### ISSUE TITLE
Is it possible to validate JSON.stringfied data sent in FormData, using interfaces? As if @Body()

### ISSUE THREAD

**Issue by yalcinozer** [lukeautry/tsoa#1027]
Generally, I'm using @Body() to receive data. But if I try to FormData [@UplodedFile() and @FormField() together ], I'm losing validation with interfaces, which makes development a lot easier. 

I spend some time looking into source code to find a way to validate data manually, in the controller.
All I want is to parse received JSON string and validate it using one of the interface, if it is possible. Thanks.

## Sorting

- **I'm submitting a ...**

  - [ ] bug report
  - [ ] feature request
  - [X] support request

- I confirm that I
  - [X] used the [search](https://github.com/lukeautry/tsoa/search?type=Issues) to make sure that a similar issue hasn't already been submit

## Expected Behavior

TSOA runs good. This is just a question.

## Context (Environment)

Version of the library: ^3.6.1
Version of NodeJS: v14.17.1

- Confirm you were using yarn not npm: [ ]

**Comment by josselinbuils** [comment]
Hello, I also need this feature and I would like to help implementing it 🙂

After some tests, it needs:
- Removing the hardcoded string type in `getFormFieldParameter` method and call `getValidatedType` instead. 
- Parsing multipart fields before reaching the request handler, using `multer` for example.

It is a working solution but looks like a hack and I'm not sure it is the best one, maybe a specific `@JsonFormField` annotation that would make tsoa parse the json, or ask users to do it themselves in the doc would be better?

**Comment by josselinbuils** [comment]
@WoH, I would like to implement that feature but I would like to be sure to go in the right direction.

Should I try to update the current `@FormField` and make it compatible with JSON or should I create a specific one, like `@JsonFormField`.

Moreover, TSOA does not seem to parse JSON itself but relies on external configuration/middlewares. Should we conserve that behaviour and ask users to parse the form fields themselves in this case (with some doc) or should TSOA manage it?

**Comment by WoH** [comment]
> Moreover, TSOA does not seem to parse JSON itself but relies on external configuration/middlewares. Should we conserve that behaviour and ask users to parse the form fields themselves in this case (with some doc) or should TSOA manage it?

Everyone should manage this on their own. I haven't thought about this in detail, but my initial goal would be that if someone correctly parses form data in a middleware before the controller mehods are invoked, validation should work with `@FormField`.

**Comment by josselinbuils** [comment]
I agree on that but if we ask users to parse the JSON themselves, they will have to call `multer.single()` in the parsing middleware in order to get the JSON fields to parse and so they won't be able to use neither  `@UploadedFile` or `@UploadedFiles` decorators.

Is it acceptable or we must find a way to be able to use them anyway?

**Comment by josselinbuils** [comment]
To provide more contexte, here is the middleware I implemented for this purpose on my side:

```ts
/**
 * Extracts files and parse text fields as JSON in multipart requests.
 */
export function multipartParserMiddleware(
  req: Request,
  res: Response,
  next: NextFunction
): void {
  if (req.header('content-type')?.includes('multipart/form-data')) {
    multer().single('file')(req, res, (error) => {
      if (error) {
        next(error);
      } else {
        try {
          Object.entries(req.body).forEach(([key, value]) => {
            req.body[key] = parseJson(key, value as string);
          });
          next();
        } catch (error) {
          next(error);
        }
      }
    });
  } else {
    next();
  }
}
```

It is very specific to my use case but it is just to show the need to call `multer`.

**Comment by josselinbuils** [comment]
One solution could be just to remove the hardcoded string type in the `getFormFieldParameter` method and use ` this.getValidatedType(parameter);` instead.

It would be flexible enough to let users parse JSON if they want to without adding any complexity.
